### PR TITLE
qa: add support/qa for cephfs-shell on CentOS 9 / RHEL9

### PR DESCRIPTION
**Pull request by batrick**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2023-11-30_08:05:19-fs:shell-wip-batrick-testing-20231130.014408-distro-default-smithi/

Failures are:
- https://tracker.ceph.com/issues/63699 (new)
- https://tracker.ceph.com/issues/63700 (new)

The second one I thought I've seen before but I cannot find an open ticket for it.

File: qa/suites/fs/shell/distro [hunk 1]
```diff
@@ -1 +1 @@
-.qa/distros/supported-random-distro$
\ No newline at end of file
+.qa/distros/supported
```
Comment (dparmar18) [inline qa/suites/fs/shell/distro:1]: `supported-random-distro$` and `supported` both point the same yamls, so is this change really needed?

Comment (batrick) [reply qa/suites/fs/shell/distro:1]: `supported-random-distro$` picks a random yaml in the directory. We want a matrix with all of the supported distros.

**Review by dparmar18 — APPROVED** [review]

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test api

File: qa/cephfs/begin/0-install.yaml [hunk 2]
```diff
@@ -3,6 +3,7 @@ tasks:
       extra_packages:
         rpm:
         - python3-cephfs
+        - cephfs-shell
```
Comment (vshankar) [inline qa/cephfs/begin/0-install.yaml:6]: @batrick I think this is causing failures with rhea 8.6 - /a/vshankar-2024-01-10_06:48:43-fs-wip-vshankar-testing-20240103.072409-testing-default-smithi/7511104

```
2024-01-10T07:08:44.407 INFO:teuthology.orchestra.run.smithi146.stderr:Error: Unable to find a match: cephfs-shell
```

ceph.spec.in conditional build cephfs-shell for reel version >= 9, so the pkg isn't available for 8.6.
